File: Java/com/leetcode/romanToInteger/RomanToInteger.java
package com.leetcode.romanToInteger;

import java.util.HashMap;

import org.omg.Messaging.SYNC_WITH_TRANSPORT;

public class RomanToInteger {
	
	public int romanToInt(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        
        HashMap<Character, Integer> table = new HashMap<Character, Integer>();
        
        table.put('I', 1);
        table.put('V', 5);
        table.put('X', 10);
        table.put('L', 50);
        table.put('C', 100);
        table.put('D', 500);
        table.put('M', 1000);
        
        int index = 0;
        int num = 0;
        while (index < s.length()) {
            int value = table.get(s.charAt(index));
            if (index + 1 < s.length() && table.get(s.charAt(index + 1)) > value) {
                num += table.get(s.charAt(++index)) - value;
            } else {
                num += value;
            }
            ++index;
        }
        
        return num;
    }

}

File: Python/roman_to_integer/roman_to_integer.py
class Solution(object):
    def romanToInt(self, s):
        d  = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000,
                        'IV': 4, 'IX': 9, 'XL': 40, 'XC': 90, 'CD': 400, 'CM': 900}
        num = 0
        idx = 0
        while idx < len(s):
            if idx < len(s) - 1:
                cs = s[idx:idx+2]
                if cs in d:
                    num += d[cs]
                    idx += 2
                    continue
            num += d[s[idx]]
            idx += 1
        return num
